### PR TITLE
Extract validations and Listener from UI Class

### PR DESCRIPTION
**Pull request by phash**
refactor SendPanel
own class for ActionListenerImplementation
Validations.class for all Validations
Tests

### CODE REVIEW

File: src/main/java/org/semux/api/validation/Validations.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+    /**
+     * validates the length of a memo message
+     * 
+     * @param memo
+     * @return true if memo is in bounds
+     */
+    public static boolean validateMemoSize(String memo) {
+        return null == memo || Bytes.of(memo).length <= Constants.MEMO_SIZE;
+    }
```
Comment (semuxproject) [inline src/main/java/org/semux/api/validation/Validations.java]: `memo` is deprecated name. Please use `data` instead

File: src/main/java/org/semux/gui/panel/SendPanel.java [hunk 2]
```diff
@@ -365,4 +302,76 @@ public String toString() {
...
+            switch (action) {
+            case REFRESH:
+                refresh();
+                break;
+            case SEND:
+                sendAction();
+                break;
+            case CLEAR:
+                clear();
+                break;
```
Comment (semuxproject) [inline src/main/java/org/semux/gui/panel/SendPanel.java]: Recommend renaming to `send()` for consistency.

File: src/main/java/org/semux/config/Constants.java [hunk 3]
```diff
@@ -88,6 +88,16 @@
...
 
+    /**
+     * the maximum size of memos
+     */
+    public static final int MEMO_SIZE = 128;
+
+    /**
+     * length of address
+     */
+    public static final int ADDRESS_LENGTH = 20;
```
Comment (semuxproject) [inline src/main/java/org/semux/config/Constants.java]: Address is already defined in `EdDSA`

File: src/main/java/org/semux/api/validation/Validations.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+    /**
+     * Validates the heigth of the fee
+     * 
+     * @param fee
+     * @param config
+     * @return
+     */
+    public static boolean validateFeeHighEnough(long fee, Config config) {
+        return fee >= config.minTransactionFee();
```
Comment (semuxproject) [inline src/main/java/org/semux/api/validation/Validations.java]: The fee validation rule will be changed soon.

**Comment by phash** [comment]
redoing the changes

**Pull request closed without merging**